tools/npm/npm-guide.md

```diff
@@ -0,0 +1,183 @@
+# Introduction
+
+### What is npm?
+
+**npm** stands for **Node Package Manager** and helps with installation and management of packages in Node.js by providing a useful command line interface.
+
+### What is a Node package?
+
+A Node package is a library of code that implements a functionality. Naturally, developers want to import and use it within a project instead of re writing it.
+
+### Where can i get the Node packages from?
+
+Users open source and publish their packages in [npm Registry](https://www.npmjs.com/search). Since these packages are open and free, developers usually check README, stats and stars of each package in [GitHub](https://www.github.com) in order to decide how useful is a library before using it.
+
+### Installing npm
+
+npm is installed along with [NodeJS](https://nodejs.org/en/download/) installation. This link provides NodeJS binaries and installers.
+
+Alternatively for linux users run the following command from terminal replacing the node version `setup_8.x`. This example is for node version 8.
```

Comment:
This is not the best way to do it. It's a lot to explain, please contact me to talk about it